tests/ragger/test_sign.py

```diff
@@ -1,67 +1,450 @@
+        "gas": GAS_LIMIT,
+        "to": ADDR2,
+        "value": 31415926913374232,
+        "chainId": 11297108109
+    }
+    common(firmware, backend, navigator, tx_params, test_name, BIP32_PATH2)
+
+
+# Legacy Zemu contract data warning: Try to blind sign with setting disabled
+def test_legacy_contract(firmware: Firmware,
+                         backend: BackendInterface,
+                         navigator: Navigator,
+                         test_name: str):
+
+    buffer = bytes.fromhex("058000002c8000003c800000010000000000000000f849208506fc23ac008303dc3194f650c3d88d12db855b8bf7d11be6c55a4e07dcc980a4a1712d6800000000000000000000000000000000000000000000000000000000000acbc7018080")
+    app_client = EthAppClient(backend)
+
+    try:
+        with app_client.send_raw(0xe0, 0x04, 0x00, 0x00, buffer):
+            if firmware.device.startswith("nano"):
+                # No screens recorded on Stax
+                moves = [
+                    NavInsID.RIGHT_CLICK,
+                    NavInsID.BOTH_CLICK
+                ]
+                navigator.navigate_and_compare(ROOT_SNAPSHOT_PATH,
+                                                test_name,
+                                                moves)
+
+    except ExceptionRAPDU as e:
+        assert e.status == StatusWord.INVALID_DATA
 
 
 def test_1559(firmware: Firmware, backend: BackendInterface, navigator: Navigator):
-    common(firmware, backend, navigator, {
+    tx_params: dict = {
         "nonce": NONCE,
         "maxFeePerGas": Web3.to_wei(145, "gwei"),
         "maxPriorityFeePerGas": Web3.to_wei(1.5, "gwei"),
         "gas": GAS_LIMIT,
         "to": ADDR,
         "value": Web3.to_wei(AMOUNT, "ether"),
         "chainId": CHAIN_ID
-    })
+    }
+    common(firmware, backend, navigator, tx_params)
+
+
+def test_sign_simple(firmware: Firmware,
+                     backend: BackendInterface,
+                     navigator: Navigator,
+                     test_name: str):
+    tx_params: dict = {
+        "nonce": NONCE2,
+        "gasPrice": Web3.to_wei(GAS_PRICE, 'gwei'),
+        "gas": GAS_LIMIT,
+        "to": ADDR2,
+        "value": 0x6f9c9e7bf61818,
+        "chainId": CHAIN_ID
+    }
+    common(firmware, backend, navigator, tx_params, test_name, "m/44'/60'/1'/0/0")
+
+
+def test_sign_limit_nonce(firmware: Firmware,
+                          backend: BackendInterface,
+                          navigator: Navigator,
+                          test_name: str):
+    tx_params: dict = {
+        "nonce": 2**64-1,
+        "gasPrice": 10,
+        "gas": 50000,
+        "to": ADDR2,
+        "value": 0x08762,
+        "chainId": CHAIN_ID
+    }
+    common(firmware, backend, navigator, tx_params, test_name, "m/44'/60'/1'/0/0")
+
+
+def test_sign_nonce_display(firmware: Firmware,
+                            backend: BackendInterface,
+                            navigator: Navigator,
+                            test_name: str):
+    # Activate nonce display
```

Comment:
This could use the `settings_toggle` [function](https://github.com/LedgerHQ/app-ethereum/blob/develop/client/src/ledger_app_clients/ethereum/settings.py#L46) from the client

Reply:
Done